stdlib/array.mli

```diff
@@ -247,6 +256,13 @@ val find_opt : ('a -> bool) -> 'a array -> 'a option
 
     @since 4.13 *)
 
+val findi_opt : (int -> 'a -> bool) -> 'a array -> (int * 'a) option
```

Comment:
This function is subsumed by the more general
```
val findi_map: (int -> 'a -> 'b option) -> 'a array -> 'b option
```
which also includes your `position_opt` as a special case.

Reply:
I think I would also be in favour of rather adding only that function to avoid inflation. Note that `base` calls it `find_mapi` though.

Reply:
Yeah, `find_mapi` should be sufficient to cover uses cases of `findi`. I'll then add `find_mapi` and revert `findi`. However, redefining `find_index` (see the below discussion) in terms of `find_mapi` is kind of annoying, so I'd prefer it as a separate function anyways.